libs/ui/packages/native/src/components/Form/DrawerTabSelector/index.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useEffect } from "react";
+import Text from "../../Text";
+import Flex from "../../Layout/Flex";
+import styled, { useTheme } from "styled-components/native";
+import { TouchableOpacity } from "react-native";
+import Animated, { useSharedValue, useAnimatedStyle, withSpring } from "react-native-reanimated";
+
+const StyledTouchableOpacity = styled(TouchableOpacity)<{ width: number }>`
+  width: ${(p) => p.width}px;
+  flex: 1;
+  height: 100%;
+`;
+
+const StyledFlex = styled(Flex)<{ isSelected: boolean }>`
+  height: 100%;
+  justify-content: center;
+  align-items: center;
+`;
+
+const StyledText = styled(Text)<{ isSelected: boolean }>`
+  line-height: 14.52px;
+  overflow: visible;
+  text-align: center;
+  font-size: 12px;
+  color: ${(p) =>
+    p.isSelected ? p.theme.colors.constant.black : p.theme.colors.opacityDefault.c50};
+`;
+
+interface OptionButtonProps<T> {
+  option: T;
+  selectedOption: T;
+  handleSelectOption: (option: T) => void;
+  label: string;
+  width: number;
+}
+
+const OptionButton = <T,>({
+  option,
+  selectedOption,
+  handleSelectOption,
+  label,
+  width,
+}: OptionButtonProps<T>) => {
+  const isSelected = selectedOption === option;
+
+  return (
+    <StyledTouchableOpacity width={width} onPress={() => handleSelectOption(option)}>
+      <StyledFlex isSelected={isSelected}>
+        <StyledText fontWeight="semiBold" isSelected={isSelected} numberOfLines={1}>
+          {label}
+        </StyledText>
+      </StyledFlex>
+    </StyledTouchableOpacity>
+  );
+};
+
+interface TabSelectorProps<T extends string | number> {
+  options: T[];
+  selectedOption: T;
+  handleSelectOption: (option: T) => void;
+  labels: { [key in T]: string };
+}
+
+export default function DrawerTabSelector<T extends string | number>({
```

Comment:
why not `TabSelector` ? i mean in term of UI it's not a drawer , am i wrong ?

Reply:
out of scope